controllers/managedocs_controller_test.go

```diff
@@ -862,9 +905,17 @@ var _ = Describe("ManagedOCS controller", func() {
 				dmsSecret := dmsSecretTemplate.DeepCopy()
 				dmsSecret.Data["SNITCH_URL"] = []byte("test-key")
 
+				// Create smtp secret with valid keys
+				smtpSecret := smtpSecretTemplate.DeepCopy()
+				smtpSecret.Data["host"] = []byte("test-host")
+				smtpSecret.Data["port"] = []byte("20")
+				smtpSecret.Data["username"] = []byte("test-key")
+				smtpSecret.Data["password"] = []byte("test-password")
```

Comment:
This and the prev 3-4 tests are about the creation of the alertmanager config under certain circumstances. 
This means that we are setting up a state for all alertmanager config pre-condition on each one of the test cases. 
This includes: 
1. pager duty secret
2. dms secret 
3. smtpSecret secret.

We already encountered a similar situation with uninstall conditions. In that case, we introduced `setupUninstallConditions` to help us manage the setup of uninstallation conditions in a central place. This allowed us to introduce new conditions without the need to adjust each test case for uninstalltion. 

My suggestion is to repeat that pattern here. We can introduce another helper:
```
func setupAlertmanagerConfigConditions(pdSecret bool, dmsSecret bool, smtpSecret bool)
```
and use it in all test cases that test for the existence of an alertmanager config